### PR TITLE
MAT-9265: QDM Test Case JSON - Deserialize QDM Entity types to their concrete implementations

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## MADiE PR Template

Jira Ticket: [MAT-9265](https://jira.cms.gov/browse/MAT-9265)
(Optional) Related Tickets:

### Summary

Configure Jackson to deserialize QDM Test Case Json QDM Entity subtypes into their implementation type.

The concrete implementations of QDM Entities (Patient, Care Partner, Practitioner, Organization, and Location) were not set as subtypes to the Entity superclass. This caused data loss on Entity subtypes when deserializing the Test Case JSON into Java models to shift test case dates.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e. Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/attributes/Entity.java [hunk 1]
```diff
@@ -13,6 +15,12 @@
 @NoArgsConstructor
 @AllArgsConstructor
 @JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.PROPERTY,
+    property = "_type",
+    visible = true)
+@JsonSubTypes({@JsonSubTypes.Type(value = Practitioner.class, name = "QDM::Practitioner")})
```
Comment (adongare) [inline src/main/java/gov/cms/madie/models/cqm/datacriteria/attributes/Entity.java]: no need to include other types?

**Review by adongare — DISMISSED** [review]
Nice catch Joe🥇

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/attributes/Entity.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/attributes/Entity.java]: Ah, man... I got excited the PoC worked and forget to add the rest. Thank you, sir!

**Review by adongare — APPROVED** [review]
